src/psij/job_executor_config.py

```diff
@@ -9,8 +9,8 @@ class JobExecutorConfig(object):
 
     DEFAULT_WORK_DIRECTORY = Path.home() / '.psij' / 'work'
 
-    def __init__(self, launcher_log_file: Optional[Path] = None,
-                 work_directory: Optional[Path] = None) -> None:
+    def __init__(self, launcher_log_file: Optional[Union[str, Path]] = None,
+                 work_directory: Optional[Union[str, Path]] = None) -> None:
```

Comment:
I would favor keeping the declared type as Path and doing the conversion as a convenience if you want something quickly and did not consult the docs or ran a typechecker.

Reply:
ack, done